pkg/kapp/cmd/kapp.go

```diff
@@ -138,6 +138,17 @@ func NewKappCmd(o *KappOptions, flagsFactory cmdcore.FlagsFactory) *cobra.Comman
 		return nil
 	})
 
+	configureTTY := func(cmd *cobra.Command) {
+		var forceTTY bool
+		// Default tty to true for deploy and delete commands: https://github.com/vmware-tanzu/carvel-kapp/issues/28
+		_, defaultTTY := cmd.Annotations["cli.carvel.dev/tty-by-default"]
+
+		cmd.Flags().BoolVar(&forceTTY, "tty", defaultTTY, "Force TTY-like output")
+		cmd.PreRun = func(cmd *cobra.Command, args []string) {
```

Comment:
instead of PreRun, lets try using cobrautil.WrapRunEForCmd (https://github.com/vmware-tanzu/carvel-kapp/blob/331c4ef932b203a4acee83b98e239a0b05d11a4d/vendor/github.com/cppforlife/cobrautil/misc.go#L33). otherwise we would have to consider what if PreRun is already set, etc...

Reply:
> otherwise we would have to consider what if PreRun is already set, etc...

Totally makes sense. I will make the change.

Reply:
Done